src/main.cpp

```diff
@@ -435,9 +435,54 @@ int main(int argc,char **argv)
   Vec lambda;
   VecCreateSeq(PETSC_COMM_SELF, optimctx->getNoptimvars() - optimctx->getNdesign(), &lambda);
   VecSetFromOptions(lambda);
-  VecSet(lambda, 1.0); // Set to one for now. TODO: Initialize somehow. 
+  /* Initial Lagrangian multiplier is zero */
+  VecSet(lambda, 0.0);
   VecAssemblyBegin(lambda);
   VecAssemblyEnd(lambda);
+  optimctx->lambda = &lambda;
+
+  bool load_optimvar = config.GetBoolParam("load_optimvar", false);
+  double old_mu;  // discontinuity penalty strength at the previous optimization iteration.
+
+  /* Set initial starting point */
+  if (load_optimvar) {
```

Comment:
I agree that using hdf5 would be better than saving binary files from one rank. For now, please change the filename extension to something unique, e.g., "bin".